### PR TITLE
Rails asociations to Private Events

### PR DESCRIPTION
**Pull request by elberthcorniell**
In this PR we want to add the use of Rails asociations to:

- Create an event with an user,
- Create attendees to an event using a user and event ids,
- Show previous and upcoming events.
- Show previous and upcoming events for users.

### CODE REVIEW

**Review by javitocor — CHANGES_REQUESTED** [review]
# STATUS CHANGES REQUIRED 🛑 

Hello, team! ✋ 
Amazing job with this project! 👍 
I can see you have worked hard here! 💯 
Congratulations! 👏 
However, we still need to implement some changes in order to get his project approved! 💪 
You have done a great job here, your code is good and it works, but it is also important to follow the requirements given, it is something you need to practice in order to work professionally when you land a job, and this is a project to practice and master the associations in rails  🤔 
Kindly check my comments below 👇  apply the changes and request a new code review ⏩ 
Good luck! 😃 
## Mandatory changes
- The upcoming and past events must be located in the models, remember MVC, in the models we manage the associations and the relationship with the database, not in the controllers. Kindly, place your upcoming and past scopes in the events model.
- Add associations between the users and the events as requested
User model: `has_many :events, foreign_key: :creator_id` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_assoc_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
Event model `belongs_to :creator, class_name: "User"` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_assoc_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
- Also associations as the user and the event attendee. Kindly check this article int he rails guides to have an idea
There are 2 equally good options to achieve that, described here https://guides.rubyonrails.org/association_basics.html#choosing-between-has-many-through-and-has-and-belongs-to-many
- In your event create action, should use #build method on association, i.e. `current_user.events.build(event_params)` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/build_method_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
- And in the index action of the events controller, as in the show action in the user controller, make use of the scope generated before
    * [x] lists past events, i.e. `@past_events = Event.previous_events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
    * [x] lists ucomping events, i.e. `@upcoming_events = Event.upcoming_events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
    * [x] lists events created by user, i.e. `@created_events = current_user.events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/current_user_show_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
    * [x] lists events previoulsy attendented by user **using event model scope**, i.e. `@past_events = current_user.attended_events.previous` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
    * [x] lists upcoming events previously attended by user **using event model scope**, i.e. `@upcoming_events = current_user.attended_events.upcoming` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
- Good job with the testing, however, one of the main requirements was to use Rspec, kindly add it to your gemfile, set it up and create your unit and integration test using Rspec
    * [x] models have unit tests for validations and associations (Rspec) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/unit_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))
    * [x] authentication have has integration tests (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))
    *[ ] event managment features have integration tests (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))

File: app/controllers/events_controller.rb [hunk 1]
```diff
@@ -0,0 +1,92 @@
+# rubocop:disable Style/GuardClause
+
+class EventsController < ApplicationController
+  before_action :set_event, only: %i[show edit update destroy]
+  # before action :user_logged_in, only: %i[show index]
```
Comment (lekegitrepo) [inline app/controllers/events_controller.rb]: - **_Remove the commented code_**

**Review by lekegitrepo — CHANGES_REQUESTED** [review]
## Status: Changes Required :recycle:
### Hi @elberthcorniell and @cvilla714 :clap: :clap:
#### Good job so far :+1: However, you need to implement the following requirements(base on the previously required changes) before this project can be approved.

#### Tests:
- **_You need to add integration tests: it's going to cover user authentication(the signup, sign-in), event management(event creation, and attending an event), and remember to test for both positive and negative scenarios. _** 

#### Commented Codes:
- **_Remove all the commented codes_**

#### _Reviewed by Temitayo Adeleke_

**Comment by elberthcorniell** [comment]
Thanks for the feedback Temitayo. We'll work on this as soon as possible! 🤜🏼🤛🏼

**Review by teekaytech — APPROVED** [review]
# Status: Approved :green_circle: 

### Congratulations team, this project has passed all the mandatory requirements. It is therefore approved :heavy_check_mark:

**[COMMENTS]**
_Although the project is approved because all mandatory requirements are satisfied, kindly make these changes._
- Update your README.md file to reflect on how to test your recently added integration tests.
- You might want to check the invite feature, there seems to be an issue when a user is invited.

Good luck with your next project.

* * * 
**Taofeek Olalere
[Twitter](https://twitter.com/ola_lere) | [Github](https://github.com/teekaytech) | @taofeekolalere on Slack**
__Happy Coding :smiley:__

**Pull request merged**